Extension/src/LanguageServer/Providers/documentFormattingEditProvider.ts

```diff
@@ -15,7 +15,7 @@ export class DocumentFormattingEditProvider implements vscode.DocumentFormatting
 
     public provideDocumentFormattingEdits(document: vscode.TextDocument, options: vscode.FormattingOptions, token: vscode.CancellationToken): Promise<vscode.TextEdit[]> {
         return new Promise<vscode.TextEdit[]>((resolve, reject) => {
-            this.client.notifyWhenReady(() => {
+            this.client.notifyWhenReady(async () => {
```

Comment:
If converting all calls to notifyWhenReady to use async lambdas, I think you can remove the override of notifyWhenReady that accepts non-async lambdas.

Reply:
in some places in client.ts, `notifyWhenReady` uses accepts normal non-async lambdas